Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(date 1487700926000)
+++ build.gradle	(revision )
@@ -5,7 +5,7 @@
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:2.3.0'
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
Index: app/src/main/java/tinkoff/androidcourse/Response.java
===================================================================
--- app/src/main/java/tinkoff/androidcourse/Response.java	(date 1487700926000)
+++ app/src/main/java/tinkoff/androidcourse/Response.java	(date 1487700926000)
@@ -1,9 +0,0 @@
-package tinkoff.androidcourse;
-
-/**
- * @author Vladimir Kokhanov
- */
-class Response {
-
-    String value;
-}
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(date 1487700926000)
+++ app/build.gradle	(revision )
@@ -10,7 +10,7 @@
         }
     }
     compileSdkVersion 25
-    buildToolsVersion "24.0.2"
+    buildToolsVersion "25.0.2"
     defaultConfig {
         applicationId "tinkoff.androidcourse"
         minSdkVersion 14
@@ -41,7 +41,8 @@
     androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
-    compile 'com.android.support:appcompat-v7:25.1.0'
-    compile 'com.google.code.gson:gson:2.4'
+    compile 'com.android.support:appcompat-v7:25.2.0'
+    compile 'com.google.code.gson:gson:2.7'
+    compile 'com.android.support:recyclerview-v7:25.2.0'
     testCompile 'junit:junit:4.12'
 }
Index: app/src/main/java/tinkoff/androidcourse/DialogsAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/tinkoff/androidcourse/DialogsAdapter.java	(revision )
+++ app/src/main/java/tinkoff/androidcourse/DialogsAdapter.java	(revision )
@@ -0,0 +1,61 @@
+package tinkoff.androidcourse;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+public class DialogsAdapter extends RecyclerView.Adapter<DialogsAdapter.ViewHolder> {
+
+    private List<DialogItem> dataset;
+    private OnItemClickListener clickListener;
+
+    public DialogsAdapter(List<DialogItem> dataset, OnItemClickListener clickListener) {
+        this.dataset = dataset;
+        this.clickListener = clickListener;
+    }
+
+    @Override
+    public DialogsAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chat_dialog, parent, false);
+        return new ViewHolder(view, clickListener);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        holder.title.setText(dataset.get(position).getTitle());
+        holder.desc.setText(dataset.get(position).getDesc());
+    }
+
+    @Override
+    public int getItemCount() {
+        return dataset.size();
+    }
+
+    public static class ViewHolder extends RecyclerView.ViewHolder {
+
+        public TextView title;
+        public TextView desc;
+
+        public ViewHolder(View view, OnItemClickListener listener) {
+            super(view);
+            title = (TextView) view.findViewById(R.id.tv_dialog_title);
+            desc = (TextView) view.findViewById(R.id.tv_dialog_desc);
+            setListener(listener);
+        }
+
+        private void setListener(final OnItemClickListener listener) {
+            itemView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (listener != null) {
+                        listener.onItemClick(getAdapterPosition());
+                    }
+                }
+            });
+        }
+    }
+}
\ No newline at end of file
Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(date 1487700926000)
+++ app/src/main/res/layout/activity_main.xml	(revision )
@@ -4,14 +4,11 @@
     android:id="@+id/activity_main"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:paddingBottom="@dimen/activity_vertical_margin"
-    android:paddingLeft="@dimen/activity_horizontal_margin"
-    android:paddingRight="@dimen/activity_horizontal_margin"
-    android:paddingTop="@dimen/activity_vertical_margin"
     tools:context="tinkoff.androidcourse.MainActivity">
 
-    <TextView
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Hello World!" />
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recycler_view_dialogs"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:scrollbars="vertical" />
 </RelativeLayout>
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(date 1487700926000)
+++ app/src/main/AndroidManifest.xml	(revision )
@@ -8,7 +8,12 @@
         android:label="@string/app_name"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+
+        <activity android:name=".MainActivity" />
+
+        <activity
+            android:name=".LoginActivity">
+
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
\ No newline at end of file
Index: app/src/main/java/tinkoff/androidcourse/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/tinkoff/androidcourse/MainActivity.java	(date 1487700926000)
+++ app/src/main/java/tinkoff/androidcourse/MainActivity.java	(revision )
@@ -2,20 +2,51 @@
 
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.widget.Toast;
 
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
+import java.util.ArrayList;
+import java.util.List;
 
 public class MainActivity extends AppCompatActivity {
 
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter adapter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        Gson gson = new GsonBuilder().create();
-        Response response = gson.fromJson("{ value:\"bla bla bla\" }", Response.class);
-        Toast.makeText(this, "Response: " + response.value, Toast.LENGTH_LONG).show();
+        initRecyclerView();
+    }
+
+    private void initRecyclerView() {
+        recyclerView = (RecyclerView) findViewById(R.id.recycler_view_dialogs);
+        recyclerView.setHasFixedSize(true);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(layoutManager);
+        adapter = new DialogsAdapter(createDataset(), new OnItemClickListener() {
+            @Override
+            public void onItemClick(int position) {
+                Toast.makeText(MainActivity.this, "position = " + position, Toast.LENGTH_SHORT).show();
+            }
+        });
+        recyclerView.setAdapter(adapter);
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(this, layoutManager.getOrientation());
+        recyclerView.addItemDecoration(dividerItemDecoration);
+    }
+
+    private List<DialogItem> createDataset() {
+        List<DialogItem> list = new ArrayList<>();
+        list.add(new DialogItem("title", "desc"));
+        list.add(new DialogItem("title", "desc"));
+        list.add(new DialogItem("title", "desc"));
+        list.add(new DialogItem("title", "desc"));
+        list.add(new DialogItem("title", "desc"));
+        list.add(new DialogItem("title", "desc"));
+        return list;
     }
 }
Index: app/src/main/java/tinkoff/androidcourse/DialogItem.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/tinkoff/androidcourse/DialogItem.java	(revision )
+++ app/src/main/java/tinkoff/androidcourse/DialogItem.java	(revision )
@@ -0,0 +1,20 @@
+package tinkoff.androidcourse;
+
+public class DialogItem {
+
+    private String title;
+    private String desc;
+
+    public DialogItem(String title, String desc) {
+        this.title = title;
+        this.desc = desc;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getDesc() {
+        return desc;
+    }
+}
Index: app/src/main/res/layout/activity_login.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_login.xml	(revision )
+++ app/src/main/res/layout/activity_login.xml	(revision )
@@ -0,0 +1,38 @@
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:gravity="center_horizontal"
+    android:orientation="vertical"
+    android:paddingBottom="@dimen/activity_vertical_margin"
+    android:paddingLeft="@dimen/activity_horizontal_margin"
+    android:paddingRight="@dimen/activity_horizontal_margin"
+    android:paddingTop="@dimen/activity_vertical_margin"
+    tools:context="tinkoff.androidcourse.LoginActivity">
+
+    <EditText
+        android:id="@+id/edit_text_login"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:hint="@string/login"
+        android:maxLines="1"
+        android:singleLine="true" />
+
+    <EditText
+        android:id="@+id/edit_text_password"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:hint="@string/password"
+        android:imeActionId="@+id/login"
+        android:imeOptions="actionUnspecified"
+        android:inputType="textPassword"
+        android:maxLines="1"
+        android:singleLine="true" />
+
+    <Button
+        android:id="@+id/btn_enter"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="32dp"
+        android:text="@string/enter" />
+</LinearLayout>
Index: app/src/main/java/tinkoff/androidcourse/LoginActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/tinkoff/androidcourse/LoginActivity.java	(revision )
+++ app/src/main/java/tinkoff/androidcourse/LoginActivity.java	(revision )
@@ -0,0 +1,39 @@
+package tinkoff.androidcourse;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+public class LoginActivity extends AppCompatActivity {
+
+    private EditText login;
+    private EditText password;
+    private Button button;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        login = (EditText) findViewById(R.id.edit_text_login);
+        password = (EditText) findViewById(R.id.edit_text_password);
+        button = (Button) findViewById(R.id.btn_enter);
+
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                startNextScreen();
+            }
+        });
+    }
+
+    private void startNextScreen() {
+        Intent intent = new Intent(this, MainActivity.class);
+        intent.putExtra("LOGIN", login.getText().toString());
+        startActivity(intent);
+    }
+}
+
Index: gradle/wrapper/gradle-wrapper.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- gradle/wrapper/gradle-wrapper.properties	(date 1487700926000)
+++ gradle/wrapper/gradle-wrapper.properties	(revision )
@@ -3,4 +3,4 @@
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
Index: app/src/main/java/tinkoff/androidcourse/OnItemClickListener.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/tinkoff/androidcourse/OnItemClickListener.java	(revision )
+++ app/src/main/java/tinkoff/androidcourse/OnItemClickListener.java	(revision )
@@ -0,0 +1,6 @@
+package tinkoff.androidcourse;
+
+public interface OnItemClickListener {
+
+    void onItemClick(int position);
+}
Index: app/src/main/res/layout/item_chat_dialog.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/item_chat_dialog.xml	(revision )
+++ app/src/main/res/layout/item_chat_dialog.xml	(revision )
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_gravity="center"
+    android:orientation="vertical"
+    android:background="?android:attr/selectableItemBackground">
+
+    <TextView
+        android:id="@+id/tv_dialog_title"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingLeft="16dp"
+        android:paddingRight="16dp"
+        android:paddingTop="8dp"
+        android:textSize="18sp"
+        tools:text="Test test test test" />
+
+    <TextView
+        android:id="@+id/tv_dialog_desc"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingBottom="8dp"
+        android:paddingLeft="16dp"
+        android:paddingRight="16dp"
+        android:textSize="14sp"
+        tools:text="Test test test test 2" />
+</LinearLayout>
\ No newline at end of file
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/strings.xml	(date 1487700926000)
+++ app/src/main/res/values/strings.xml	(revision )
@@ -1,3 +1,7 @@
 <resources>
     <string name="app_name">AndroidCourseDebug</string>
+
+    <string name="login">Логин</string>
+    <string name="password">Пароль</string>
+    <string name="enter">Войти</string>
 </resources>
